lms/validation/_lti_launch_params.py

```diff
@@ -14,9 +15,30 @@ class LaunchParamsSchema(PyramidRequestSchema):
     """
 
     resource_link_id = fields.Str(required=True)
+    launch_presentation_return_url = fields.URL()
+
+    # If we have an error in one of these fields we should redirect to the LTI
+    # app if possible
+    lti_redirect_fields = {"resource_link_id"}
 
     locations = ["form"]
 
+    # The parent params are all specified with typing stuff and are pretty
+    # crazy, so we'll disable the warnings to get slightly easier to understand
+    # params
```

Comment:
This comment confused me for a while. I wasn't sure what the "parent" being referred to was and didn't realise that this comment had anything to do with/was explaining the reason for the `# pylint` comment below.

You might want to update the comment: I don't think the triggering of the pylint warning has anything to do with typing. And I don't think marshmallow's params are particularly crazy here, if you remove the typing they can be written as just `(self, error, data, *, many, **kwargs)` and that won't trigger a pylint warning.

The problem that's triggering the pylint warning seems to be that you have to keep the `many` split out, you can't merge it into `kwargs`, which may be an issue with pylint:

* * *

You can avoid the pylint warning without using any typing, like this:

```python
def handle_error(self, error, data, *, _many, **kwargs):
```

But, I think the fact that your current signature `(self, error, data, **kwargs)` triggers the pylint warning may be a bug in pylint. I may be wrong, but aren't `**kwargs` keyword-only arguments? So just `**kwargs` is the same as `*, _many, **kwargs)`. Could be wrong. But I think pylint might be wrong here.

Reply:
I don't really know, what `*` is doing here. You can't have `_many` because that is different too. However you can have:
```
def handle_error(self, error, data, *, many, **kwargs)
```
As long as you pass many onto your call to super it seems happy.

Reply:
`_many` is ok, pylint will allow that.

The `*` enforces that `many` / `_many` can _only_ be used as a keyword argument, never as a positional argument